Rock/Model/AttendanceCodeService.Partial.cs

```diff
@@ -158,40 +152,25 @@ public static AttendanceCode GetNew( int alphaNumericLength, int alphaLength, in
         /// <returns></returns>
         public static string GetNextNumericCodeAsString( int alphaNumericLength, int alphaLength, int numericLength, bool isRandomized, string lastCode )
         {
-            // Find a good unique numeric code for today
-            string numericCode = string.Empty;
-
             if ( isRandomized )
             {
-                numericCode = GenerateRandomNumericCode( numericLength );
-                // #2877, use contains to prevent leading zeros bypassing a match for 666
-                while ( noGood.Any( s => numericCode.Contains( s ) ) || _todaysCodes.Any( c => c.EndsWith( numericCode ) ) )
-                {
-                    numericCode = GenerateRandomNumericCode( numericLength );
-                }
+                return GenerateRandomNumericCode( numericLength );
             }
-            else
-            {
-                if ( !string.IsNullOrEmpty( lastCode ) )
-                {
-                    var maxCode = lastCode.Substring( alphaNumericLength + alphaLength );
-                    int nextCode = maxCode.AsInteger() + 1;
 
-                    // Let's just skip over this one...
-                    if ( nextCode.ToString().Contains( "666" ) )
-                    {
-                        nextCode += 1;
-                    }
+            if ( !string.IsNullOrEmpty( lastCode ) )
+            {
+                var maxCode = lastCode.Substring( alphaNumericLength + alphaLength );
+                int nextCode = maxCode.AsInteger() + 1;
 
-                    numericCode = nextCode.ToString( "D" + numericLength );
-                }
-                else
+                // Let's just skip over this one...
+                if ( nextCode.ToString().Contains( "666" ) )
```

Comment:
I'm still reviewing all these code changes, but if you're going to skip number codes this way, you'll need to include 411, and 911 too.

Reply:
This part of the code is actually totally broken, I reported the problem to Jon. Chose not to try to fix it as part of this PR, though.

Reply:
It seems inconsistent to check for specific strings here when the `NoGood` array already contains all the other combinations to avoid (including 666 and 911).

Reply:
I believe this is going to be changed to use 'contains' on the NoGood code checking.

SIDE NOTE:  Also, I'm creating new tests in the test project for testing AttendanceCodes -- and future PRs will also need to use-- and add new tests if necessary.  For now, after a dev runs the tests, they will need to have the 'Fact Skip' added back in so we don't break anyone's CI environment that is not set up with a db for testing purposes.  Later on, we'll figure out a way to keep these tests un-skipped so they will be run for everyone who's running the tests in CI.

Reply:
@chrismcgrath I'm now ready to try out your code (tomorrow) since we now have new tests (via 7d864d706756ab9deb17249a871e337d73b0bb49) for various edge cases.  If you want to run these tests in your dev environment, you can check out the new readme  (via 71d96e0c6f46a13d2f01bb2ab926f7b727a08937).